### ISSUE TITLE
Grafana 9.0.0 403 forbidden when accessing provisoned prometheus datasource

### ISSUE THREAD

**Issue by jonaslar** [grafana/grafana#51163]
**What happened**: After Grafana was upgraded to version 9.0.0, we get 403-forbidden responses on requests to links like these in the grafana web gui. The two examples are from the Explore view, but we get the same in panels using the same provisioned prometheus datasources:

- https://<replaced>/api/datasources/13/resources/api/v1/rules
- https://<replaced>/api/datasources/13/resources//api/v1/query_exemplars?query=test&start=1655811456220&end=1655813256220

![image](https://user-images.githubusercontent.com/25543712/174796014-1ccc19ca-f179-4121-90ac-9a9af2a80ec5.png)

This only happens for the provisioned Prometheus datasources we have declared like this:

``` yaml
apiVersion: 1

datasources:
- name: icpdq02
  orgId: 2
  type: prometheus
  access: proxy
  url: https://thanos.openshift.domain.no
  jsonData:
    httpHeaderName1: 'Authorization'
  secureJsonData:
    httpHeaderValue1: 'Bearer secretRemovedByMe'
```

**What you expected to happen**: No 403 responses and able to query the prometheus datasources

**How to reproduce it (as minimally and precisely as possible)**:
Log in to Grafana Web UI. Navigate to Explorer view and choose a prometheus datasource that is provisioned.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.0.0
- Data source type & version: Prometheus
- OS Grafana is installed on:  Red Hat Enterprise Linux 7.9
- User OS & Browser: Windows and Chrome
- Grafana plugins:
- Others:
Running query directly with curl and with the same HTTP-header Authorization  works. Like this:

``` bash
curl --request GET \
  --url 'https://thanos.openshift.domain.no/api/v1/query?query=cluster%3Amemory_usage%3Aratio&dedup=true&partial_response=false&time=1655878927.793' \
  --header 'Authorization: Bearer secretRemovedByMe' 

{
	"status": "success",
	"data": {
		"resultType": "vector",
		"result": [
			{
				"metric": {
					"__name__": "cluster:memory_usage:ratio",
					"prometheus": "openshift-monitoring/k8s"
				},
				"value": [
					1655878927.793,
					"0.34632231621099124"
				]
			}
		]
	}
}
```  
  
I changed the prometheus url from https to http and ran tcpdump on grafana host. I can not see that the Autorization header specified in the datasource is being sent:

``` bash
55164069 ecr 1706375853], length 219: HTTP: GET /api/v1/series?end=1655891797&match%5B%5D=cgroup_cpuacct_usage&start=1655888197 HTTP/1.1
E...1.@.@......;...&.*.Ppr.u.k.^...:q......
h...e.:.GET /api/v1/series?end=1655891797&match%5B%5D=cgroup_cpuacct_usage&start=1655888197 HTTP/1.1
Host: thanos.openshift.domain.no
User-Agent: Grafana/9.0.0
Accept-Encoding: gzip


12:13:31.026418 IP api.icpdq02.norsk-tipping.no.http > p1grfn001.prod01.domain.no.44330: Flags [.], seq 1:1449, ack 219, win 59, options [nop,nop,TS val 1706375860 ecr 175516406$], length 1448: HTTP: HTTP/1.1 403 Forbidden
```

**Comment by jonaslar** [comment]
Hi, this issue was solved by applying patch 9.0.1.

**Comment by jonaslar** [comment]
Hi, this issue was solved by applying patch 9.0.1.
